File: Exercism/fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let productionRate = 221.
let productionRatePerHour =
    function
    | s when s > -1 && s <= 4 -> (s |> float) * productionRate * 1.
    | s when s > -1 && s <= 8 -> (s |> float) * productionRate * 0.9
    | s when s > -1 && s <= 9 -> (s |> float) * productionRate * 0.8
    | s when s > -1 && s <= 10 -> (s |> float) * productionRate * 0.77
    | s -> failwith $"Invalid rate {s}"

let workingItemsPerMinute speed =
    (productionRatePerHour speed / 60.)
    |> floor
    |> int

File: Exercism/csharp/cars-assemble/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed) => speed switch {
      <= 0 => 0.0,
      <= 4 => 1.0,
      <= 8 => 0.9,
      <= 9 => 0.8,
      _ => 0.77
    };

    public static double ProductionRatePerHour(int speed) => 221.0 * speed * SuccessRate(speed);
    public static int WorkingItemsPerMinute(int speed) => (int)ProductionRatePerHour(speed) / 60;
}